includes/plugin-compat.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Plugin compatibility functionality
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace { // Ensure global namespace.
+	use Google\Web_Stories\Story_Post_Type;
+
+	if ( ! function_exists( '\is_amp_endpoint' ) && is_singular( Story_Post_Type::POST_TYPE_SLUG ) ) {
```

Comment:
Had to do it this way, otherwise it failed in the following scenario:

1. Web Stories plugin is active
1. Function does not exist, and is declared here
1. AMP plugin is being activated
1. Fatal error because it tries to declare the function

Reply:
@westonruter @spacedmonkey thoughts on this approach for the polyfill?

Reply:
The `is_singular()` check is happening too early, no? It can't be called until the `wp` action whereas currently it is running at `plugins_loaded`. So you could either replace the `plugins_loaded` action with the `wp` action, or you could remove the `is_singular()` conditional the outside and move it inside to `return is_singular( Story_Post_Type::POST_TYPE_SLUG );`

Reply:
Moving it inside doesn‘t work, for the reason stated above.

Reply:
Ah, on the `wp` action I could indeed do that.